### PR TITLE
Add second table to facilitate request level sanitized queries and qu…

### PR DESCRIPTION
**Pull request by chelseatroy**
…eries passed through allow list plus merged with telemetry data

## Apropos of t[his conversation in JIRA](https://mozilla-hub.atlassian.net/browse/DSRE-661?focusedCommentId=557036) on the bigquery error and [this conversation on GitHub](https://github.com/mozilla/bigquery-etl/pull/3012#discussion_r890463622) about one table versus two, I have made the changes (I think) to do two tables:

1. `merino_log_sanitized_v3`: For just the request-level sanitized queries
2. `suggest_impression_sanitized_v3`: Honestly here I just copied what's happening in the v2 table by the same name and changed all the v2 references to v3. 

Please let me know what more there is to do here!

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/metadata.yaml
Comment (quiiver) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/metadata.yaml:9]:
```suggestion

```

I believe we can remove the bit about it being passed through the keyword list

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:14]: This and similar references should be changed to `moz-fx-data-shared-prod.search_terms_derived.merino_log_sanitized_v3`.

In addition, since the sanitizer service won't be writing to this table anymore (airflow will) the sanitizer should have its write access removed via `metadata.yaml`.

**Comment by rebecca-burwei** [comment]
When this is resolved, please let DS know what the final changes were, so that we are using the correct BQ tables for analysis. Either a slack message in the contextual-services-data channel or announcement at the weekly team meeting on Thursdays would be fine.

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/metadata.yaml
Comment (chelseatroy) [reply sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/metadata.yaml:9]: Can do!

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (chelseatroy) [reply sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:14]: Done and removed!

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (quiiver) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:1]: This query needs to be a join of the impressions table and the sanitized output it's very similar to this here https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v2_external/query.sql

With the exception of filtering against the know keyword list in the `remotesettings_suggestions_v1` table.

In psuedo code it's something like:

```sql

with impressions as (
  select 
  -- a bunch of stuff from the merino impressions table
  from `contextual_services_stable.quicksuggest_impression_v1`
  where date(submission_date) = @submission_date
), 
sanitized_queries as (
  select *
  from `search_terms_derived.merino_log_sanitized_v3`
  where date(timestamp) = @submission_date
)
select * 
from sanitized_queries
join impressions using (request_id)
```

**Review by quiiver — APPROVED** [review]
lgtm!

File: sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/init.sql
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/init.sql:2]:
```suggestion
  `moz-fx-data-shared-prod.search_terms_derived.merino_log_sanitized_v3`(
```

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:42]: It looks like this was just copy pasted from elsewhere but I thought that we required all tables to be fully qualified. I'm not going to block merging this PR on this but I thought I would mention it.

File: sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml:11]: This is missing BQ time partitioning present in `init.sql` e.g.

```
bigquery:
  time_partitioning:
    field: submission_timestamp
    type: day
    require_partition_filter: true
    expiration_days: 15
```

**Review by whd — CHANGES_REQUESTED** [review]
A couple more changes and I think this is ready to merge

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:42]: Can we include a safety check similar to https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v2/query.sql for making sure that the merino log v3 partition has data for the particular day?

This will alert us on the DE side that there's a failure. Alternatively I suppose a DAG could read the job state from the job metadata table to determine whether to proceed.

Comment (chelseatroy) [reply sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:42]: Change pushed!

File: sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:49]: Seeing `Unrecognized name: sanitized_query; Did you mean sanitized_queries? at [42:11]`, guessing
```suggestion
    COUNT(query) AS _n_with_query,
```
is the fix.

Comment (chelseatroy) [reply sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v3/query.sql:49]: D'oh! Yep. That's what it's called in that table.

File: sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml
Comment (whd) [reply sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml:11]: this still appears to be missing from `merino_log_sanitized_v3`, though I see it was added to `suggest_impression_sanitized_v3` (where it was also missing but was meant to be added from https://github.com/mozilla/bigquery-etl/pull/3012#discussion_r890457882)

Comment (jklukas) [reply sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml:11]:
```suggestion
bigquery:
  time_partitioning:
    field: timestamp
    type: day
    require_partition_filter: true
    expiration_days: 15
workgroup_access:
```

File: sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/search_terms_derived/merino_log_sanitized_v3/metadata.yaml:10]:
```suggestion
  A sanitized variant of search terms submitted to Firefox.

  This table allows removal of PII and longer retention than
  the underlying logs.
  
  See https://github.com/mozilla/search-terms-sanitization (private)
  for the code that populates this table.
```

**Comment by chelseatroy** [comment]
Oh my gosh, we did it!!

**Pull request merged**
